DMCompiler/DM/Visitors/DMObjectBuilder.cs

```diff
@@ -134,10 +158,48 @@ class DMObjectBuilder {
             }
         }
 
+        // TODO Move this to an appropriate location
+        static public IEnumerable<DMASTProcStatement> GetStatements(DMASTProcBlockInner block) {
+            foreach (var stmt in block.Statements) {
+                yield return stmt;
+                List<DMASTProcBlockInner> recurse;
+                switch (stmt) {
+                    case DMASTProcStatementSpawn ps: recurse = new() { ps.Body }; break;
+                    case DMASTProcStatementIf ps: recurse = new() { ps.Body, ps.ElseBody }; break;
+                    case DMASTProcStatementFor ps: recurse = new() { ps.Body }; break;
+                    case DMASTProcStatementForLoop ps: recurse = new() { ps.Body }; break;
+                    case DMASTProcStatementWhile ps: recurse = new() { ps.Body }; break;
+                    case DMASTProcStatementDoWhile ps: recurse = new() { ps.Body }; break;
+                    // TODO Good luck if you declare a static var inside a switch
+                    case DMASTProcStatementSwitch ps: {
+                            recurse = new();
+                            foreach (var swcase in ps.Cases) {
+                                recurse.Add(swcase.Body);
+                            }
+                            break;
+                        }
+                    case DMASTProcStatementTryCatch ps: recurse = new() { ps.TryBody, ps.CatchBody }; break;
+                    default: recurse = new(); break;
+                }
+                foreach (var subblock in recurse) {
+                    if (subblock == null) { continue; }
+                    foreach (var substmt in GetStatements(subblock)) {
+                        yield return substmt;
+                    }
+                }
+            }
+        }
+
         private void SetVariableValue(DMVariable variable, DMASTExpression value, DMValueType valType = DMValueType.Anything) {
             DMExpression expression = DMExpression.Create(_currentObject, variable.IsGlobal ? DMObjectTree.GlobalInitProc : null, value, variable.Type);
             expression.ValType = valType;
 
+            if (variable.IsGlobal) {
+                variable.Value = new Expressions.Null();
+                EmitInitializationAssign(variable, expression);
+                return;
+            }
+
```

Comment:
This was a workaround because the const checking wasn't picking up stuff like dereferences, (world.timeofday in the test case). The main issue is that failure of const evaluation probably shouldnt raise any errors unless the const keyword is specified. Either way that is out of the scope of this PR and I think is handled differently in the open const PR so I will remove this bit of code.